lib/msf/util/exe.rb

```diff
@@ -671,6 +671,31 @@ def self.to_win64pe_service(framework, code, opts = {})
     exe_sub_method(code,opts)
   end
 
+  # self.set_template_default_winpe_dll
+  #
+  # Set the default winpe DLL template. It will select the template based on the parameters provided including the size
+  # architecture and an optional flavor. See data/templates/src/pe for template source code and build tools.
+  #
+  # @param opts [Hash]
+  # @param arch The architecture, as one the predefined constants.
```

Comment:
```suggestion
  # @param arch [String] The architecture, which will be one of the predefined architecture constants (ex ARCH_X64).
```

Reply:
I left the type out here intentionally because it should be an opaque constant. Developers should use the constant and not `'x86'` incase it changes in the future. The value of the constant could change in the future.

Reply:
Ideally I think it would be nice if we could define some wrapper class that sits around this opaque constant to avoid the potential for YARD to end up skipping this function when generating documentation but I think that is likely a discussion for another PR.